### ISSUE TITLE
OverflowError on Python2.7

### ISSUE THREAD

**Issue by georg90** [munawarb/Python-Kill-Thread-Extension#4]
This will give an `OverflowError` signed integer is greater than maximum in Python2.7

Any hints?

**Comment by munawarb** [comment]
Hmm, I'm wondering if we can change it to use some sort of Big Integer class. The problem is the thread ID is considered to be an opaque object as far as C is concerned, so it only has issues in that module. I'll take a look and see what I can come up with. Thanks for the heads-up! Have you tried it using Python 3? Do you get the same problem?

**Comment by georg90** [comment]
Haven't tried Python 3 (the HW is only running on 2.7) - but then we would use the other code sample anyway, right? 

We thought about using a float for the ident? I am really not into C at all..

**Comment by munawarb** [comment]
What hardware are you running it on? It could be an issue with word size. Did you build the module on the system in question or did you build on a different machine and then transfer it over?

**Comment by georg90** [comment]
Running it on a Pepper robot from Softbanks robotics. Will try to compile on the robot itself. It was compiled on MacOS, so this could be an issue.

**Comment by munawarb** [comment]
Ok. Let me know how it goes. Since you compiled it on a Mac and are using it on a micro processor this could definitely be the issue because the size of an integer is probably smaller on your hardware than it is on the Mac. If this doesn't resolve it, I'll see what I can do. Some sample code would help as well.

**Comment by georg90** [comment]
I tried compiling on the robot, but it fails to have the necessayr ressources like gcc. I think there is no easy way to install those anyway.

I compiled in a Linux VM using the same python version and got the same result: `except signed integer is greater than maximum`

**Comment by munawarb** [comment]
Ok. Can you give me some sample code that you ran on Linux?

**Comment by georg90** [comment]
What kind of sample code? 

I use the Thread implementation as above and then start my websocket connection
```
self.recognize_thread = CustomThread(function=self.recognize_using_websocket)
self.recognize_thread.start()
```
I then want to kill `self.recognize_thread` using `threader`. I am doing this by calling `self.recognize_thread.end()` but this returns the above error.

**Comment by georg90** [comment]
We kind off got around this issue, but it still bugs me. Great library, though!

**Comment by munawarb** [comment]
Hi,
Can you try grabbing the latest master branch and compiling the module on Linux again? I was using int to represent the thread ID in the C module but Python uses unsigned long to represent the thread ID, so this could be causing the overflow in your case.

**Comment by munawarb** [comment]
Great! I'm glad you were able to come up with an alternative. When you 
get a chance try testing the latest update of master and let me know how 
it works for you. It's bugging me as well, but reading more into how 
Python is handling its thread identifiers might be cluing me into a 
hidden problem in the Threader module that's never shown itself before. :)

**Comment by owneroxxor** [comment]
Hi there! I am having the same issue

Python 3.7.0

```
    threader.killThread(self.ident)
OverflowError: signed integer is greater than maximum
```
Have you guys came up with a solution to this unsigned long from python Thread ident overflowing the limit on the C side?

**Comment by munawarb** [comment]
Hi @owneroxxor, what architecture are you running the module on?

**Comment by owneroxxor** [comment]
Nvm, the defuncts are my fault.

**Comment by munawarb** [comment]
Ok, thanks for the update.

**Comment by munawarb** [comment]
Hi @georg90 and @owneroxxor,
This issue recently came up again and I did some of my own digging and found the issue. I've fixed it in #5. If you're still using this module, can you pull from Master and compile the latest code?
